### ISSUE TITLE
Per environment flags with js_of_ocaml

### ISSUE THREAD

**Issue by pbiggar** [ocaml/dune#1613]
I want to use different js_of_ocaml flags in my project, depending on whether we are in dev or release profile.

Dune [already adds](https://github.com/ocaml/dune/blob/55531596991f0f2d3b60f24183ea0eb3f80c2fd6/src/js_of_ocaml_rules.ml#L12) `--pretty` and `--source-maps-inline` in dev, however, there are more jsoo options to play with, and I get dramatically better debugging with the [recommended settings](https://ocsigen.org/js_of_ocaml/dev/manual/options):

```
For Debugging: "–pretty –no-inline –debug-info" + eventually "–disable staticeval –disable share"
For Production: "–opt 3". It minimize the generated javascript by applying various optimizations until a fix-point is reached
```

I tried using an `env` block, but couldn't make it work any way I tried it. Is there an existing way to do this, and if not:
- can a way to do this be added?
- can we pick better defaults for the dev and production env?

**Comment by Lupus** [comment]
I'm trying to get exception backtraces with jsoo by using flags `--enable=with-js-error --enable=excwrap`, but exception is being thrown by Lwt, and it seems that adding these flags to my binary does not change the way Lwt is being built, and thus I don't get the backtrace attached to the exception... Isn't there a way to set those globally? Or they have to be set within corresponding environment so that propagate to all jsoo invocations?

**Comment by hhugo** [comment]
Fixed by #5049
